### ISSUE TITLE
[package] ffmpeg/6.1: Failed to build for Android

### ISSUE THREAD

**Issue by arttet** [conan-io/conan-center-index#23729]
### Description

Failed to build "ffmpeg" for Android

ffmpeg/6.1: **ERROR: aom >= 1.0.0 not found using pkg-config**

### Package and Environment Details

* Package: **ffmpeg/6.1**
* Operating System: **Android**
* Compiler: **Clang 17**
* Conan version: **conan 2.2.3**
* Python version: **Python 3.12**
* Android NDK: **r26c**

### Conan profile

```text
[settings]
arch=armv8
os=Android
os.api_level=32

compiler=clang
compiler.version=17
compiler.libcxx=c++_static
compiler.cppstd=17

[options]

# ffmpeg: https://github.com/conan-io/conan-center-index/tree/master/recipes/ffmpeg/all

# Workaround: https://github.com/conan-io/conan-center-index/issues/20511
ffmpeg/*:with_libfdk_aac=False

# Workaround: https://github.com/conan-io/conan-center-index/issues/23718
ffmpeg/*:with_libx264=False

# Workaround: https://github.com/conan-io/conan-center-index/issues/23721
ffmpeg/*:with_libx265=False

# Workaround: https://github.com/conan-io/conan-center-index/issues/23724
ffmpeg/*:with_openh264=False

# Workaround: https://github.com/conan-io/conan-center-index/issues/23726
ffmpeg/*:with_libsvtav1=False

# Workaround: https://github.com/conan-io/conan-center-index/issues/23728
ffmpeg/*:with_asm=False

# https://docs.conan.io/2/reference/config_files/global_conf.html
[conf]
tools.cmake.cmaketoolchain:generator=Ninja

[tool_requires]
android-ndk/r26c
```

### Steps to reproduce

[Here](https://github.com/arttet/reproduce-bugs/tree/bug/conan-android-ffmpeg-aom) is a minimal example to reproduce the bug.

### Logs

<details><summary>Click to expand log</summary>

```
-------- Installing package ffmpeg/6.1 (27 of 27) --------
ffmpeg/6.1: Building from source
ffmpeg/6.1: Package ffmpeg/6.1:1c26447c430cfe5eba3a3aded11bdb6dfff67b40
ffmpeg/6.1: Copying sources to build folder
ffmpeg/6.1: Building your package in /home/runner/.conan2/p/b/ffmpe5503884589807/b
ffmpeg/6.1: Calling generate()
ffmpeg/6.1: Generators folder: /home/runner/.conan2/p/b/ffmpe5503884589807/b/build-debug/conan
ffmpeg/6.1: Generating aggregated env files
ffmpeg/6.1: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
ffmpeg/6.1: Calling build()
ffmpeg/6.1: apply_conandata_patches(): No patches defined in conandata
ffmpeg/6.1: Calling:
 > "/home/runner/.conan2/p/b/ffmpe5503884589807/b/src/configure" --disable-shared --enable-static --prefix=/ '--bindir=${prefix}/bin' '--libdir=${prefix}/lib' --pkg-config-flags=--static --disable-doc --enable-cross-compile --disable-asm --disable-shared --enable-static --enable-pic --enable-avdevice --enable-avcodec --enable-avformat --enable-swresample --enable-swscale --enable-postproc --enable-avfilter --enable-bzlib --enable-zlib --enable-lzma --enable-iconv --enable-libopenjpeg --disable-libopenh264 --enable-libvorbis --enable-libopus --disable-libzmq --disable-sdl2 --disable-libx264 --disable-libx265 --enable-libvpx --enable-libmp3lame --disable-libfdk-aac --enable-libwebp --enable-libaom --enable-openssl --disable-alsa --disable-libpulse --disable-vaapi --disable-vdpau --disable-libxcb --disable-libxcb-shm --disable-libxcb-shape --disable-libxcb-xfixes --disable-appkit --disable-avfoundation --disable-coreimage --disable-audiotoolbox --disable-videotoolbox --disable-securetransport --disable-cuda --disable-cuvid --enable-nonfree --enable-gpl --disable-vulkan --disable-libsvtav1 --enable-libdav1d --arch=aarch64 --disable-optimizations --disable-mmx --disable-stripping --enable-debug --nm=/home/runner/.conan2/p/andro0cd2cd114dfd6/p/bin/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-nm --ar=/home/runner/.conan2/p/andro0cd2cd114dfd6/p/bin/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-ar --strip=/home/runner/.conan2/p/andro0cd2cd114dfd6/p/bin/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-strip --cc=/home/runner/.conan2/p/andro0cd2cd114dfd6/p/bin/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android32-clang --cxx=/home/runner/.conan2/p/andro0cd2cd114dfd6/p/bin/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android32-clang++ --ld=/home/runner/.conan2/p/andro0cd2cd114dfd6/p/bin/toolchains/llvm/prebuilt/linux-x86_64/bin/ld --ranlib=/home/runner/.conan2/p/andro0cd2cd114dfd6/p/bin/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-ranlib --target-os=android '--extra-cflags=-fPIC --sysroot /home/runner/.conan2/p/andro0cd2cd114dfd6/p/bin/toolchains/llvm/prebuilt/linux-x86_64/sysroot -g' '--extra-ldflags=--sysroot /home/runner/.conan2/p/andro0cd2cd114dfd6/p/bin/toolchains/llvm/prebuilt/linux-x86_64/sysroot' 
ffmpeg/6.1: RUN: "/home/runner/.conan2/p/b/ffmpe5503884589807/b/src/configure" --disable-shared --enable-static --prefix=/ '--bindir=${prefix}/bin' '--libdir=${prefix}/lib' --pkg-config-flags=--static --disable-doc --enable-cross-compile --disable-asm --disable-shared --enable-static --enable-pic --enable-avdevice --enable-avcodec --enable-avformat --enable-swresample --enable-swscale --enable-postproc --enable-avfilter --enable-bzlib --enable-zlib --enable-lzma --enable-iconv --enable-libopenjpeg --disable-libopenh264 --enable-libvorbis --enable-libopus --disable-libzmq --disable-sdl2 --disable-libx264 --disable-libx265 --enable-libvpx --enable-libmp3lame --disable-libfdk-aac --enable-libwebp --enable-libaom --enable-openssl --disable-alsa --disable-libpulse --disable-vaapi --disable-vdpau --disable-libxcb --disable-libxcb-shm --disable-libxcb-shape --disable-libxcb-xfixes --disable-appkit --disable-avfoundation --disable-coreimage --disable-audiotoolbox --disable-videotoolbox --disable-securetransport --disable-cuda --disable-cuvid --enable-nonfree --enable-gpl --disable-vulkan --disable-libsvtav1 --enable-libdav1d --arch=aarch64 --disable-optimizations --disable-mmx --disable-stripping --enable-debug --nm=/home/runner/.conan2/p/andro0cd2cd114dfd6/p/bin/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-nm --ar=/home/runner/.conan2/p/andro0cd2cd114dfd6/p/bin/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-ar --strip=/home/runner/.conan2/p/andro0cd2cd114dfd6/p/bin/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-strip --cc=/home/runner/.conan2/p/andro0cd2cd114dfd6/p/bin/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android32-clang --cxx=/home/runner/.conan2/p/andro0cd2cd114dfd6/p/bin/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android32-clang++ --ld=/home/runner/.conan2/p/andro0cd2cd114dfd6/p/bin/toolchains/llvm/prebuilt/linux-x86_64/bin/ld --ranlib=/home/runner/.conan2/p/andro0cd2cd114dfd6/p/bin/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-ranlib --target-os=android '--extra-cflags=-fPIC --sysroot /home/runner/.conan2/p/andro0cd2cd114dfd6/p/bin/toolchains/llvm/prebuilt/linux-x86_64/sysroot -g' '--extra-ldflags=--sysroot /home/runner/.conan2/p/andro0cd2cd114dfd6/p/bin/toolchains/llvm/prebuilt/linux-x86_64/sysroot' 
ERROR: aom >= 1.0.0 not found using pkg-config

If you think configure made a mistake, make sure you are using the latest
version from Git.  If the latest version fails, report the problem to the
ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.libera.chat.
Include the log file "ffbuild/config.log" produced by configure as this will help
solve the problem.

ffmpeg/6.1: ERROR: 
Package '1c26447c430cfe5eba3a3aded11bdb6dfff67b40' build failed
ffmpeg/6.1: WARN: Build folder /home/runner/.conan2/p/b/ffmpe5503884589807/b/build-debug
ERROR: ffmpeg/6.1: Error in build() method, line 710
	autotools.configure()
	ConanException: Error 1 while executing
```

</details>

**Comment by arttet** [comment]
Duplicate of https://github.com/conan-io/conan-center-index/issues/23009

**Comment by elvisdukaj** [comment]
Can you try this fix: https://github.com/conan-io/conan-center-index/pull/23368

**Comment by arttet** [comment]
Hi @elvisdukaj,

I have checked that fix. Unfortunately, the issue is still actual.

```sh

ERROR: aom >= 1.0.0 not found using pkg-config

If you think configure made a mistake, make sure you are using the latest
version from Git.  If the latest version fails, report the problem to the
ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.libera.chat.
Include the log file "ffbuild/config.log" produced by configure as this will help
solve the problem.

ffmpeg/6.1@demo/testing: ERROR:
```

**Comment by uilianries** [comment]
Hello @arttet and @elvisdukaj  We just merged the PR #24566 that should mitigate Android build for FFmpeg. Please, update your local copy with `conan install -r conancenter --update --requires=ffmpeg/<version>` and try it.

The current Android build logs can be found here: https://github.com/conan-io/conan-center-index/pull/24566#issuecomment-2292915235

Regards!

**Comment by elvisdukaj** [comment]
Hi @uilianries  I'll test locally and let you know soon! Thank you for the great work!
